contentcuration/contentcuration/models.py

```diff
@@ -1658,9 +1658,13 @@ def recalculate_editors_storage(self):
     def on_create(self):
         self.changed = True
         self.recalculate_editors_storage()
+        if self.published:
+            delete_public_channel_cache_keys()
 
     def on_update(self):
         self.changed = self.changed or self.has_changes()
+        if self.published:
+            delete_public_channel_cache_keys()
```

Comment:
For changes to the `on_create` and `on_update` methods of `ContentNode`, I don't think this is exactly what we want. The `published` attribute is only really used on the root node of the channel tree, foreign key referenced on the channel as `main_tree`. We want to clear the cache:
1) When a public channel is published
1) When a published channel is made public
2) When a public channel is made private

You've already handled (2) and (3), but these changes don't exactly handle (1) because the change wasn't necessarily that the _channel_ was being published. Luckily, there's a [single function](https://github.com/learningequality/studio/blob/unstable/contentcuration/contentcuration/utils/publish.py#L810) through which channels are published and this `published` attribute is maintained for the root node. I think that's a better location for handling case (1)

Reply:
Commits landing ASAP.